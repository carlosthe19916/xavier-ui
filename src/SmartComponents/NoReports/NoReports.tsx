import React from 'react';
import { Link } from 'react-router-dom';
import {
    Bullseye,
    EmptyStateBody,
    EmptyStateVariant,
    EmptyState,
    Title
} from '@patternfly/react-core';
import { RouterGlobalProps } from '../../models/router';
import ProcessImprovementSvg from '../../PresentationalComponents/Icons/process-improvement.svg';
import './Reports.scss';

interface StateToProps {
}

interface DispatchToProps {
}

interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
};

interface State {
};

class NoReports extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
    }

    render() {
        return (
            <Bullseye>
                <EmptyState variant={ EmptyStateVariant.full }>
                    <div className="pf-c-empty-state__icon" style={ { opacity: 0.6 } }>
                        <ProcessImprovementSvg height="80px" />
                    </div>
                    <Title size="lg">Let Red Hat Migration Analytics suggest ways to optimize your environment</Title>
                    <EmptyStateBody>
                                See how you can optimize your virtual environment by
                                uploading a Red Hat CloudForms generated inventory file
                                and then letting Red Hat Migration Analytics create
                                reports showing how you can save money and optimize
                                workloads - whether by migrating virtual machines or
                                migrating application to RHEL.
                    </EmptyStateBody>
                    <Link to={ '/reports/upload' } className="pf-c-button pf-m-primary">Get Started</Link>
                </EmptyState>
            </Bullseye>
        );
    }
}

export default NoReports;
