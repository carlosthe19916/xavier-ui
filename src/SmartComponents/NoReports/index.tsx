import { GlobalState } from '../../models/state';
import * as uploadActions from '../../actions/UploadActions';
import NoReports from './NoReports';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';

const mapStateToProps = (state: GlobalState)  => {
    const {
        uploadState: {
            file
        }
    } = state;
    return {
        file
    };
};

const mapDispatchToProps = {
    selectUploadFile: uploadActions.selectUploadFile
};

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(NoReports)
);
