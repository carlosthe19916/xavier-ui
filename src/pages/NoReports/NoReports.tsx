import React from 'react';
import { Link, Redirect } from 'react-router-dom';
import {
    Button,
    ButtonVariant,
    EmptyState,
    EmptyStateVariant,
    EmptyStateIcon,
    Title,
    EmptyStateBody,
    Bullseye,
    TitleLevel
} from '@patternfly/react-core';
import ReportsPage from '../../PresentationalComponents/ReportsPage';
import {
    AddCircleOIcon
} from '@patternfly/react-icons';
import { User } from '../../models';
import { RouterGlobalProps } from '../../models/router';

interface StateToProps {
    user: User
}

interface DispatchToProps {
}

interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
};

interface State {
};

class NoReports extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
    }

    render() {
        const { user } = this.props;

        if (user.firstTimeCreatingReports) {
            return <Redirect to="/getting-started" />;
        }

        return (
            <ReportsPage>
                <Bullseye>
                    <EmptyState variant={ EmptyStateVariant.large }>
                        <EmptyStateIcon icon={ AddCircleOIcon } />
                        <Title headingLevel={ TitleLevel.h5 } size="lg">No reports found</Title>
                        <EmptyStateBody>
                            Reports are created from inventory data files that are uploaded to Red Hat Migration Analytics.
                        </EmptyStateBody>
                        <Button variant={ ButtonVariant.primary } component={ Link } to="/reports/upload">Create report</Button>
                    </EmptyState>
                </Bullseye>
            </ReportsPage>
        );
    }
}

export default NoReports;
