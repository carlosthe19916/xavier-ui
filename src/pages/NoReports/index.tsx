import { GlobalState } from '../../models/state';
import { bindActionCreators } from 'redux';
import * as uploadActions from '../../actions/UploadActions';
import NoReports from './NoReports';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';

const mapStateToProps = (state: GlobalState)  => {
    let {
        userState: {
            user
        }
    } = state;
    return {
        user
    };
};

const mapDispatchToProps = (dispatch: any) =>
    bindActionCreators({
        selectUploadFile: uploadActions.selectUploadFile
    }, dispatch);

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(NoReports)
);
