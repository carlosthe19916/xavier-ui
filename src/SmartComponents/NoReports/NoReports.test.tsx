import { shallow } from "enzyme";
import NoReports from "./NoReports";
import Dropzone from "react-dropzone";

const props = {
  file: null,
  selectUploadFile: jest.fn(),
  match: null,
  history: {
    push: jest.fn()
  },
  location: null,
};

describe("NoReports", () => {

  it("expect to render", () => {
    const wrapper = shallow(
      <NoReports { ...props } />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render drop", () => {
    const fileContents = "file contents";
    const file = new Blob([fileContents], { type: "text/plain" });

    const wrapper = shallow(
      <NoReports { ...props } />
    );

    wrapper.find(Dropzone).simulate("drop", { dataTransfer: { files: [file] } });
    
    expect(props.selectUploadFile).toHaveBeenCalledTimes(1);
    expect(props.history.push).toHaveBeenCalledTimes(1);
  });

});
