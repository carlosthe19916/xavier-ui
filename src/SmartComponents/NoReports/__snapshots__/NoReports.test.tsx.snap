// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoReports expect to render 1`] = `
<Bullseye
  className=""
  component="div"
>
  <EmptyState
    className=""
    variant="full"
  >
    <div
      className="pf-c-empty-state__icon"
      style={
        Object {
          "opacity": 0.6,
        }
      }
    >
      <Icon
        height="80px"
      />
    </div>
    <Title
      size="lg"
    >
      Let Red Hat Migration Analytics suggest ways to optimize your environment
    </Title>
    <EmptyStateBody
      className=""
    >
      See how you can optimize your virtual environment by uploading a Red Hat CloudForms generated inventory file and then letting Red Hat Migration Analytics create reports showing how you can save money and optimize workloads - whether by migrating virtual machines or migrating application to RHEL.
    </EmptyStateBody>
    <Dropzone
      accept={
        Array [
          "application/zip",
          "application/json",
        ]
      }
      multiple={false}
      noClick={true}
      noKeyboard={true}
      onDrop={[Function]}
    >
      <Component />
    </Dropzone>
  </EmptyState>
</Bullseye>
`;
